internal/controllers/nmc_reconciler.go

```diff
@@ -565,94 +565,69 @@ func (h *nmcReconcilerHelperImpl) UpdateNodeLabelsAndRecordEvents(ctx context.Co
 	nodeModuleReadyLabels := sets.New[types.NamespacedName]()
 	deprecatedNodeModuleReadyLabels := sets.New[string]()
 
-	for label := range node.GetLabels() {
-		if ok, namespace, name := utils.IsKernelModuleReadyNodeLabel(label); ok {
-			nodeModuleReadyLabels.Insert(types.NamespacedName{Namespace: namespace, Name: name})
-		}
-
-		if utils.IsDeprecatedKernelModuleReadyNodeLabel(label) {
-			deprecatedNodeModuleReadyLabels.Insert(label)
-		}
-	}
+	getKernelModuleReadyLabels(&nodeModuleReadyLabels, &deprecatedNodeModuleReadyLabels, node)
 
 	// get spec labels and their config
 	specLabels := make(map[types.NamespacedName]kmmv1beta1.ModuleConfig)
-	for _, module := range nmc.Spec.Modules {
-		specLabels[types.NamespacedName{Namespace: module.Namespace, Name: module.Name}] = module.Config
-	}
+	getSpecLabelsAndTheirConfigs(specLabels, nmc)
 
 	// get status labels and their config
 	statusLabels := make(map[types.NamespacedName]kmmv1beta1.ModuleConfig)
-	for _, module := range nmc.Status.Modules {
-		label := types.NamespacedName{Namespace: module.Namespace, Name: module.Name}
-		statusLabels[label] = module.Config
-	}
+	getStatusLabelsAndTheirConfigs(statusLabels, nmc)
 
 	unloaded := make([]types.NamespacedName, 0, len(nodeModuleReadyLabels))
 	loaded := make([]types.NamespacedName, 0, len(specLabels))
 
 	patchFrom := client.MergeFrom(node.DeepCopy())
 
 	// label in node but not in spec or status - should be removed
-	for nsn := range nodeModuleReadyLabels {
-		_, inSpec := specLabels[nsn]
-		_, inStatus := statusLabels[nsn]
-		if !inSpec && !inStatus {
-			meta.RemoveLabel(
-				&node,
-				utils.GetKernelModuleReadyNodeLabel(nsn.Namespace, nsn.Name),
-			)
-
-			unloaded = append(unloaded, nsn)
-		}
-	}
+	unloaded = removeOrphanedLabels(nodeModuleReadyLabels, specLabels, statusLabels, unloaded, &node)
 
 	// v1 ready labels, deprecated - should be removed
 	for label := range deprecatedNodeModuleReadyLabels {
 		meta.RemoveLabel(&node, label)
 	}
 
 	// label in spec and status and config equal - should be added
-	for nsn, specConfig := range specLabels {
-		statusConfig, ok := statusLabels[nsn]
-		if ok && reflect.DeepEqual(specConfig, statusConfig) && !nodeModuleReadyLabels.Has(nsn) {
-			meta.SetLabel(
-				&node,
-				utils.GetKernelModuleReadyNodeLabel(nsn.Namespace, nsn.Name),
-				"",
-			)
-
-			loaded = append(loaded, nsn)
-		}
-	}
+	loaded = addEqualLabels(nodeModuleReadyLabels, specLabels, statusLabels, loaded, &node)
 
 	if err := h.client.Patch(ctx, &node, patchFrom); err != nil {
 		return fmt.Errorf("could not patch node: %v", err)
 	}
 
-	for _, nsn := range unloaded {
-		h.recorder.AnnotatedEventf(
-			&node,
-			map[string]string{"module": nsn.String()},
-			v1.EventTypeNormal,
-			"ModuleUnloaded",
-			"Module %s unloaded from the kernel",
-			nsn.String(),
-		)
+	h.recordModuleEvents(&node, unloaded, "unload")
+	h.recordModuleEvents(&node, loaded, "load")
```

Comment:
@yevgeny-shnaidman @TomerNewman How about we extract the events stuff to the caller of `UpdateNodeLabelsAndRecordEvents` instead of having 1 function doing "multiple different things?